src/wallet/wallet.cpp

```diff
@@ -844,51 +844,7 @@ bool CWallet::AddToWallet(const CWalletTx& wtxIn, bool fFlushOnClose)
         wtx.nTimeReceived = GetAdjustedTime();
         wtx.nOrderPos = IncOrderPosNext(&walletdb);
         wtxOrdered.insert(make_pair(wtx.nOrderPos, TxPair(&wtx, (CAccountingEntry*)0)));
-
-        wtx.nTimeSmart = wtx.nTimeReceived;
-        if (!wtxIn.hashUnset())
```

Comment:
Yes, the change replaces this and other uses of `wtxIn` with `wtx`. Because this code is running in the case where `fInsertedNew ` is true, `wtx` at this point is just a copy of `wtxIn` with two extra fields set (`nTimeReceived` and `nOrderPos`)